x-pack/plugins/endpoint/server/index_pattern.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Logger, LoggerFactory, RequestHandlerContext } from 'kibana/server';
+import { ESIndexPatternService } from '../../ingest_manager/server';
+import { EndpointAppConstants } from '../common/types';
+
+export interface IndexPatternRetriever {
+  getIndexPattern(ctx: RequestHandlerContext, datasetPath: string): Promise<string>;
+  getEventIndexPattern(ctx: RequestHandlerContext): Promise<string>;
+  getMetadataIndexPattern(ctx: RequestHandlerContext): Promise<string>;
+}
+
+export class IngestIndexPatternRetriever implements IndexPatternRetriever {
```

Comment:
would you mind adding a doc comment to this class like:

```
/**
 * explaining why this exists or when to use it
 */
export class IngestIndexPatternRetriever implements IndexPatternRetriever {
```

These doc comments will show up in tooltips in supported editing environments and it makes it easier for me to understand whats going on